### ISSUE TITLE
Caching and # nolint

### ISSUE THREAD

**Issue by krlmlr** [jimhester/lintr#68]
The `# nolint` comments don't seem to work properly with caching. They are respected only on first execution (presumably, when the cache is dirty) -- from the second execution onwards, errors are reported.

Try

```
lintr::lint_package(".", cache = TRUE)
```

for [my package](https://github.com/krlmlr/kimisc).

**Comment by jimhester** [comment]
Probably all that needs to change is changing this return to `return(exclude(lints, ...))`

https://github.com/jimhester/lintr/blob/9b5a6b275898f8a852da8c946ac9ca70cf81cd5c/R/lint.R#L45

**Comment by krlmlr** [comment]
Works for me: all existing tests pass, ignores are respected with caching. I'll submit a pull request.
